File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.py
from typing import List

class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        n = len(nums)

        # Find indices of min and max
        min_index = nums.index(min(nums))
        max_index = nums.index(max(nums))

        left = min(min_index, max_index)
        right = max(min_index, max_index)

        # 1. Remove both from the front
        front = right + 1

        # 2. Remove both from the back
        back = n - left

        # 3. Remove one from each side
        front_back = (left + 1) + (n - right)

        return min(front, back, front_back)

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.rs
impl Solution {
    pub fn minimum_deletions(nums: Vec<i32>) -> i32 {
        let n = nums.len();

        // Find indices of min and max
        let min_index = nums.iter().enumerate().min_by_key(|&(_, &v)| v).unwrap().0;
        let max_index = nums.iter().enumerate().max_by_key(|&(_, &v)| v).unwrap().0;

        let left = min_index.min(max_index);
        let right = min_index.max(max_index);

        // 1. Remove both from the front
        let front = right + 1;

        // 2. Remove both from the back
        let back = n - left;

        // 3. Remove one from each side
        let front_back = (left + 1) + (n - right);

        front.min(back).min(front_back) as i32
    }
}